tests/test_frame.py

```diff
@@ -1505,27 +1504,31 @@ def test_frame_scalars_slice() -> None:
 
     # Note: bool_ cannot be tested since the index is object and pandas does not
     # support boolean access using loc except when the index is boolean
-    check(assert_type(df.loc[str_], pd.Series), pd.Series)
+    with pytest.warns(FutureWarning, match="Comparison of Timestamp with datetime"):
```

Comment:
If we want to treat deprecation/future-warnings on main the same we do for the released pandas versions, we could change the type annotations to make mypy/pyright fail in these cases and make them negative tests.

Reply:
I agree with this.  We should go through the FutureWarnings after the next release and remove problematic code.